be2-scala/src/main/scala/ch/epfl/pop/storage/DbActor.scala

```diff
@@ -199,6 +200,31 @@ final case class DbActor(
     }
   }
 
+  //generates the key of the RollCallData to store in the database
+  private def generateRollCallDataKey(laoId: Hash): String = {
+    s"${ROLL_CALL_DATA_PREFIX}${laoId.toString}"
+  }
+
+  @throws [DbActorNAckException]
+  private def readRollCallData(laoId: Hash): RollCallData = {
+    Try(storage.read(s"${ROLL_CALL_DATA_PREFIX}${laoId.toString}")) match {
+      case Success(Some(json)) => RollCallData.buildFromJson(json)
+      case Success(None) => throw DbActorNAckException(ErrorCodes.SERVER_ERROR.id, s"ReadElectionData for election $laoId not in the database")
+      case Failure(ex) => throw ex
+    }
+  }
+
+  @throws [DbActorNAckException]
+  private def writeRollCallData(laoId: Hash, message: Message): Unit = {
+    this.synchronized {
+      val rollcallData: RollCallData = Try(readRollCallData(laoId)) match {
+        case Success(data) => data
+        case Failure(_) => RollCallData(null, null)
+      }
```

Comment:
ℹ️ In general, never use `null`  as "filler parameters" in scala ^^. If you really need to do this, I'd suggest Options or dummy values (which are not null)

Note: I believe that it's ok in this case since you create it and then fill it directly no matter what happens